test/libsolidity/syntaxTests/revertStatement/revert_event.sol

```diff
@@ -0,0 +1,8 @@
+contract C {
+    event E();
+    function f() public pure {
+        revert E();
```

Comment:
I was wondering why pure functions are allowed to revert. I think that should be forbidden.

Reply:
This has been discussed a lot in the past. Pure means the result only depends on the arguments, and that is the case here.